src/lib/Syntax.hs

```diff
@@ -784,11 +798,15 @@ instance BindsUVars UPat' where
 instance HasUVars UDecl where
   freeUVars (ULet _ p expr) = freeUVars p <> freeUVars expr
   freeUVars (UData (UConDef _ bs) dataCons) = freeUVars $ Abs bs dataCons
+  freeUVars (UInterface _ _ _) = mempty  -- TODO
+  freeUVars (UInstance _ _) = mempty     -- TODO
 
 instance BindsUVars UDecl where
   boundUVars decl = case decl of
-   ULet _ (p,_) _ -> boundUVars p
-   UData tyCon dataCons -> boundUVars tyCon <> foldMap boundUVars dataCons
+    ULet _ (p,_) _ -> boundUVars p
+    UData tyCon dataCons -> boundUVars tyCon <> foldMap boundUVars dataCons
+    UInterface _ _ _ -> mempty
+    UInstance _ _    -> mempty
```

Comment:
This seems insufficient too

Reply:
Yeah, but not declaring them seems wrong?

Reply:
I'm not sure what the alternative is. We treat instance and interfaces as always-in-scope anyway.